ddl/rollingback.go

```diff
@@ -118,6 +120,50 @@ func rollingbackAddColumn(t *meta.Meta, job *model.Job) (ver int64, err error) {
 	return ver, errCancelledDDLJob
 }
 
+func rollingbackDropSchema(t *meta.Meta, job *model.Job) error {
+	dbInfo, err := t.GetDatabase(job.SchemaID)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if dbInfo == nil {
+		job.State = model.JobStateCancelled
+		return infoschema.ErrDatabaseDropExists.GenWithStackByArgs("")
+	}
+
+	if dbInfo.State == model.StatePublic {
+		job.State = model.JobStateCancelled
+		return errCancelledDDLJob
+	}
+	// To simplify the rollback logic, cannot be canceled after job start to run.
+	// Normally won't fetch here, because there is check when cancel ddl jobs. see function: isJobRollbackable.
+	job.State = model.JobStateRunning
+	return nil
+}
+
+func rollingbackDropTable(t *meta.Meta, job *model.Job) error {
+	schemaID := job.SchemaID
+	tblInfo, err := getTableInfo(t, job, schemaID)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if tblInfo == nil {
+		job.State = model.JobStateCancelled
+		return errors.Trace(infoschema.ErrTableNotExists.GenWithStackByArgs(
+			fmt.Sprintf("(Schema ID %d)", schemaID),
+			fmt.Sprintf("(Table ID %d)", job.TableID),
+		))
```

Comment:
Please remove this code. I think we have done this logical in `getTableInfo`.

Reply:
done.